src/ethereum/EtherscanButton/index.tsx

```diff
@@ -22,21 +21,30 @@ const getNetwork = (network: Network) => {
 
 const EtherscanButton = ({
   className,
-  type,
   value,
   network = 'mainnet',
 }: Props): React.ReactElement => {
-  const etherscanLink = `https://${getNetwork(
-    network
-  )}etherscan.io/${type}/${value}`;
+  const type = value.length > 42 ? 'tx' : 'address';
+
+  const onClick = (
+    event:
+      | React.MouseEvent<HTMLAnchorElement>
+      | React.KeyboardEvent<HTMLAnchorElement>
+  ): void => {
+    event?.stopPropagation();
+    window.open(
+      `https://${getNetwork(network)}etherscan.io/${type}/${value}`,
+      '_blank'
+    );
+  };
 
   return (
     <StyledLink
       className={className}
       aria-label="Show details on Etherscan"
-      href={etherscanLink}
       rel="noopener noreferrer"
-      target="_blank">
+      onClick={onClick}
+      onKeyPress={onClick}>
```

Comment:
After finally being able to test it locally, this is what better fits the requirements: https://gist.github.com/fernandomg/1b4fcaaad70f87903be6e375f15a4d12

Reply:
@fernandomg I changed the code as you suggested, can you review again, please?